scipy/stats/_continuous_distns.py

```diff
@@ -4225,6 +4225,61 @@ def _stats(self, a, moments='mvsk'):
 
 skewnorm = skew_norm_gen(name='skewnorm')
 
+
+class trap_gen(rv_continuous):
+    """A trapezoidal continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The trapezoidal distribution can be represented with an up-sloping line
+    from ``loc`` to ``(loc + c*scale)``, then constant to ``(loc + d*scale)``
+    and then downsloping from ``(loc + d*scale)`` to ``(loc+scale)``.
+
+    `trap` takes ``c`` and ``d`` as shape parameters.
+
+    %(after_notes)s
+
+    The standard form is in the range [0, 1] with c the mode.
+    The location parameter shifts the start to `loc`.
+    The scale parameter changes the width from 1 to `scale`.
+
+    %(example)s
+
+    """
+    def _rvs(self, c, d):
+        rands = self._random_state.rand(self._size)
+        return self._ppf(rands, c, d)
+
+    def _argcheck(self, c, d):
+        return (c >= 0) & (c <= 1) & (d >= 0) & (d <= 1) & (d >= c)
+
+    def _pdf(self, x, c, d):
+        u = 2 / (d - c + 1)
+
+        condlist = [x==1, x < c, x <= d, x > d]
+        choicelist = [0, u * x / c, u, u * (1 - x) / (1 - d)]
+        return np.select(condlist, choicelist)
```

Comment:
Ok, I see that `triang` has the same edge case: c=0 is explicitly allowed despite a division by zero.